### PR TITLE
refac(contracts): apply SafeCall in the withdraw() of the ValidatorPool

### PR DESCRIPTION
**Pull request by Pangssu**
## Changes
- apply SafeCall in the window() of the ValidatorPool
- optimize storage slot of Types.Challenge

### CODE REVIEW

File: packages/contracts/contracts/libraries/Types.sol [hunk 1]
```diff
@@ -143,25 +143,25 @@ library Types {
...
      * @custom:field segStart   The L2 block number of the first segment.
      * @custom:field segSize    The number of L2 blocks.
-     * @custom:field timeoutAt  Timeout timestamp of the next turn.
      * @custom:field outputRoot The L2 output root to be replaced.
-     * @custom:field approved   Whether the challenge was approved.
      */
     struct Challenge {
-        uint256 turn;
+        uint8 turn;
+        uint64 timeoutAt;
```
Comment (kangsorang) [inline packages/contracts/contracts/libraries/Types.sol:157]: It would be nice to have the variables and comments in the right order.

**Review by kangsorang — COMMENTED** [review]
LGTM

**Review by chokobole — APPROVED** [review]
LGTM

File: packages/contracts/contracts/L1/Colosseum.sol [hunk 2]
```diff
@@ -473,9 +473,9 @@ contract Colosseum is Initializable, Semver {
      */
     function _updateTimeout(Types.Challenge storage _challenge) private {
         if (!_isAbleToBisect(_challenge)) {
-            _challenge.timeoutAt = block.timestamp + PROVING_TIMEOUT;
+            _challenge.timeoutAt = uint64(block.timestamp + PROVING_TIMEOUT);
         } else {
-            _challenge.timeoutAt = block.timestamp + BISECTION_TIMEOUT;
+            _challenge.timeoutAt = uint64(block.timestamp + BISECTION_TIMEOUT);
```
Comment (seolaoh) [inline packages/contracts/contracts/L1/Colosseum.sol:478]: is it possible to sum `uint64` and `uint256` directly like in line 637?

**Review by seolaoh — APPROVED** [review]
LGTM

**Pull request merged**